core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -110,6 +110,7 @@ private[spark] class SerializerManager(
   private def shouldCompress(blockId: BlockId): Boolean = {
     blockId match {
       case _: ShuffleBlockId => compressShuffle
+      case _: ArrayShuffleBlockId => compressShuffle
```

Comment:
will `ArrayShuffleBlockId` go through network?

Reply:
No, but `ShuffleBlockFetcherIterator` needs uncompress to detect corrupt.
```
          var isStreamCopied: Boolean = false
          try {
            input = streamWrapper(arrayBlockId, in)
```
But I think I can use below to avoid this.
```
          var isStreamCopied: Boolean = false
          try {
            input = streamWrapper(arrayBlockId.blockIds.head, in)
```